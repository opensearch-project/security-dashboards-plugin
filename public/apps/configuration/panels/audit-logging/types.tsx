/*
 *   Copyright OpenSearch Contributors
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

export interface GeneralSettings {
  // layer
  enable_rest?: boolean;
  enable_transport?: boolean;
  disabled_rest_categories?: string[];
  disabled_transport_categories?: string[];

  // attribute
  resolve_bulk_requests?: boolean;
  resolve_indices?: boolean;
  log_request_body?: boolean;
  exclude_sensitive_headers?: boolean;

  // ignore
  ignore_users?: string[];
  ignore_requests?: string[];
}

export interface ComplianceSettings {
  enabled?: boolean;

  // read
  read_metadata_only?: boolean;
  read_ignore_users?: string[];
  read_watched_fields?: {
    [key: string]: string;
  };

  // write
  write_metadata_only?: boolean;
  write_log_diffs?: boolean;
  write_ignore_users?: string[];
  write_watched_indices?: string[];
}

export interface AuditLoggingSettings {
  enabled?: boolean;
  audit?: GeneralSettings;
  compliance?: ComplianceSettings;
}
