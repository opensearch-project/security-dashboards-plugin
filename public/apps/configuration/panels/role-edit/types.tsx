/*
 *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import { EuiComboBoxOptionOption } from '@elastic/eui';

export type ComboBoxOptions = EuiComboBoxOptionOption[];
export type FieldLevelSecurityMethod = 'exclude' | 'include';
export interface RoleIndexPermissionStateClass {
  indexPatterns: ComboBoxOptions;
  docLevelSecurity: string;
  fieldLevelSecurityMethod: FieldLevelSecurityMethod;
  fieldLevelSecurityFields: ComboBoxOptions;
  maskedFields: ComboBoxOptions;
  allowedActions: ComboBoxOptions;
}

export enum TenantPermissionType {
  None = '',
  Read = 'r',
  Write = 'w',
  Full = 'rw',
}

export interface RoleTenantPermissionStateClass {
  tenantPatterns: ComboBoxOptions;
  permissionType: TenantPermissionType;
}
