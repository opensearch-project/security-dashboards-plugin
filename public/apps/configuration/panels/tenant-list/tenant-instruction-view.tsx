/*
 *   Copyright OpenSearch Contributors
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import React from 'react';
import { EuiSpacer, EuiText, EuiTitle } from '@elastic/eui';

export function TenantInstructionView() {
  return (
    <>
      <EuiTitle size="l">
        <h1>Tenants</h1>
      </EuiTitle>

      <EuiSpacer size="xxl" />

      <EuiText textAlign="center">
        <h2>You have not enabled multi tenancy</h2>
      </EuiText>

      <EuiText textAlign="center" size="xs" color="subdued" className="instruction-text">
        Contact your administrator to enable multi tenancy.
      </EuiText>
    </>
  );
}
