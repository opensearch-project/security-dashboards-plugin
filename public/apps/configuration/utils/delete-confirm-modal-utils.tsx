/*
 *   Copyright OpenSearch Contributors
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import React, { useState } from 'react';
import { EuiOverlayMask, EuiConfirmModal } from '@elastic/eui';

/**
 *
 * @param handleDelete: [Type: func] delete function which needs to be execute on click of confirm button.
 * @param entity: e.g. role(s), tenant(s), user(s), mapping etc. This will display in confirmation text before deletion.
 * @param customConfirmationText: If you want other than default confirm message, pass it as customConfirmationText.
 */
export function useDeleteConfirmState(
  handleDelete: () => Promise<void>,
  entity: string,
  customConfirmationText?: React.ReactNode
): [() => void, React.ReactNode] {
  const [isDeleteConfirmModalVisible, setIsDeleteConfirmModalVisible] = useState(false);
  const closeDeleteConfirmModal = () => setIsDeleteConfirmModalVisible(false);
  const showDeleteConfirmModal = () => setIsDeleteConfirmModalVisible(true);

  const handleConfirm = async () => {
    await handleDelete();
    closeDeleteConfirmModal();
  };

  let deleteConfirmModal;
  if (isDeleteConfirmModalVisible) {
    deleteConfirmModal = (
      <EuiOverlayMask>
        <EuiConfirmModal
          title="Confirm Delete"
          onCancel={closeDeleteConfirmModal}
          onConfirm={handleConfirm}
          cancelButtonText="Cancel"
          confirmButtonText="Confirm"
          defaultFocusedButton="confirm"
        >
          {customConfirmationText ? (
            customConfirmationText
          ) : (
            <p>Do you really want to delete selected {entity}?</p>
          )}
        </EuiConfirmModal>
      </EuiOverlayMask>
    );
  }
  return [showDeleteConfirmModal, deleteConfirmModal];
}
