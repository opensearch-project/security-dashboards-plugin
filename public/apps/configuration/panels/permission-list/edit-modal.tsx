/*
 *   Copyright OpenSearch Contributors
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import {
  EuiButton,
  EuiButtonEmpty,
  EuiModal,
  EuiModalBody,
  EuiModalFooter,
  EuiModalHeader,
  EuiModalHeaderTitle,
  EuiOverlayMask,
  EuiForm,
  EuiComboBox,
} from '@elastic/eui';
import React, { useState } from 'react';
import { ComboBoxOptions, Action } from '../../types';
import {
  stringToComboBoxOption,
  comboBoxOptionToString,
  appendOptionToComboBoxHandler,
} from '../../utils/combo-box-utils';
import { FormRow } from '../../utils/form-row';
import { NameRow } from '../../utils/name-row';

interface PermissionEditModalDeps {
  groupName: string;
  action: Action;
  allowedActions: string[];
  optionUniverse: ComboBoxOptions;
  handleClose: () => void;
  handleSave: (groupName: string, allowedActions: string[]) => Promise<void>;
}

const TITLE_DICT: { [key: string]: string } = {
  [Action.create]: 'Create new action group',
  [Action.edit]: 'Edit action group',
  [Action.duplicate]: 'Duplicate action group',
};

export function PermissionEditModal(props: PermissionEditModalDeps) {
  const [groupName, setGroupName] = useState<string>(props.groupName);
  const [allowedActions, setAllowedActions] = useState<ComboBoxOptions>(
    props.allowedActions.map(stringToComboBoxOption)
  );
  const [isFormValid, setIsFormValid] = useState<boolean>(true);

  return (
    <EuiOverlayMask>
      <EuiModal onClose={props.handleClose}>
        <EuiModalHeader>
          <EuiModalHeaderTitle>{TITLE_DICT[props.action]}</EuiModalHeaderTitle>
        </EuiModalHeader>

        <EuiModalBody>
          <EuiForm>
            <NameRow
              headerText="Name"
              headerSubText="Enter a unique name to describe the purpose of this group. You cannot change the name after creation."
              resourceName={groupName}
              resourceType=""
              action={props.action}
              setNameState={setGroupName}
              setIsFormValid={setIsFormValid}
            />
            <FormRow headerText="Permissions">
              <EuiComboBox
                options={props.optionUniverse}
                selectedOptions={allowedActions}
                onCreateOption={appendOptionToComboBoxHandler(setAllowedActions, [])}
                onChange={setAllowedActions}
              />
            </FormRow>
          </EuiForm>
        </EuiModalBody>

        <EuiModalFooter>
          <EuiButtonEmpty onClick={props.handleClose}>Cancel</EuiButtonEmpty>

          <EuiButton
            id="submit"
            onClick={async () => {
              await props.handleSave(groupName, allowedActions.map(comboBoxOptionToString));
            }}
            fill
            disabled={!isFormValid}
          >
            {props.action === Action.create ? 'Create' : 'Save'}
          </EuiButton>
        </EuiModalFooter>
      </EuiModal>
    </EuiOverlayMask>
  );
}
