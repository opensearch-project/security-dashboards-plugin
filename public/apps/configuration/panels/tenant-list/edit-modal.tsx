/*
 *   Copyright OpenSearch Contributors
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import {
  EuiButton,
  EuiButtonEmpty,
  EuiModal,
  EuiModalBody,
  EuiModalFooter,
  EuiModalHeader,
  EuiModalHeaderTitle,
  EuiOverlayMask,
  EuiForm,
  EuiTextArea,
  EuiHorizontalRule,
} from '@elastic/eui';
import React, { useState } from 'react';
import { Action } from '../../types';
import { FormRow } from '../../utils/form-row';
import { NameRow } from '../../utils/name-row';

interface TenantEditModalDeps {
  tenantName: string;
  tenantDescription: string;
  action: Action;
  handleClose: () => void;
  handleSave: (tenantName: string, tenantDescription: string) => Promise<void>;
}

const TITLE_DICT: { [key: string]: string } = {
  [Action.create]: 'Create tenant',
  [Action.edit]: 'Edit tenant',
  [Action.duplicate]: 'Duplicate tenant',
};

export function TenantEditModal(props: TenantEditModalDeps) {
  const [tenantName, setTenantName] = useState<string>(props.tenantName);
  const [tenantDescription, setTenantDescription] = React.useState<string>(props.tenantDescription);

  const [isFormValid, setIsFormValid] = useState<boolean>(true);

  const handleTenantDescriptionChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
    setTenantDescription(e.target.value);
  };

  return (
    <EuiOverlayMask>
      <EuiModal onClose={props.handleClose}>
        <EuiModalHeader>
          <EuiModalHeaderTitle>{TITLE_DICT[props.action]}</EuiModalHeaderTitle>
        </EuiModalHeader>

        <EuiModalBody>
          <EuiForm>
            <NameRow
              headerText="Name"
              headerSubText="Specify a descriptive and unique tenant name that is easy to recognize. You cannot edit the name once the tenant is created."
              resourceName={tenantName}
              resourceType="tenant"
              action={props.action}
              setNameState={setTenantName}
              setIsFormValid={setIsFormValid}
              fullWidth={true}
            />
            <FormRow
              headerText="Description"
              headerSubText="Describe the purpose of the tenant."
              optional
            >
              <EuiTextArea
                data-test-subj="tenant-description"
                fullWidth
                placeholder="Describe the tenant"
                value={tenantDescription}
                onChange={handleTenantDescriptionChange}
              />
            </FormRow>
          </EuiForm>
        </EuiModalBody>
        <EuiHorizontalRule margin="xs" />
        <EuiModalFooter>
          <EuiButtonEmpty onClick={props.handleClose}>Cancel</EuiButtonEmpty>

          <EuiButton
            id="submit"
            onClick={async () => {
              await props.handleSave(tenantName, tenantDescription);
            }}
            fill
            disabled={!isFormValid}
          >
            {props.action === Action.create ? 'Create' : 'Save'}
          </EuiButton>
        </EuiModalFooter>
      </EuiModal>
    </EuiOverlayMask>
  );
}
