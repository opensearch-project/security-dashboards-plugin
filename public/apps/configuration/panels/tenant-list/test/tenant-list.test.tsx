/*
 *   Copyright OpenSearch Contributors
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import { ManageTab } from '../manage_tab';
import { shallow } from 'enzyme';
import React from 'react';
import { EuiInMemoryTable } from '@elastic/eui';
import { useDeleteConfirmState } from '../../../utils/delete-confirm-modal-utils';
import { Tenant } from '../../../types';
import { TenantEditModal } from '../edit-modal';
import { TenantInstructionView } from '../tenant-instruction-view';
import { getDashboardsInfo } from '../../../../../utils/dashboards-info-utils';

jest.mock('../../../utils/tenant-utils');
jest.mock('../../../../../utils/auth-info-utils');
jest.mock('../../../utils/delete-confirm-modal-utils', () => ({
  useDeleteConfirmState: jest.fn().mockReturnValue([jest.fn(), '']),
}));
jest.mock('../../../utils/context-menu', () => ({
  useContextMenuState: jest
    .fn()
    .mockImplementation((buttonText, buttonProps, children) => [children, jest.fn()]),
}));
jest.mock('../../../utils/toast-utils', () => ({
  useToastState: jest.fn().mockReturnValue([[], jest.fn(), jest.fn()]),
  getSuccessToastMessage: jest.fn(),
  createUnknownErrorToast: jest.fn(),
}));

jest.mock('../../../../../utils/dashboards-info-utils', () => ({
  getDashboardsInfo: jest.fn().mockImplementation(() => {
    return mockDashboardsInfo;
  }),
}));

const mockDashboardsInfo = {
  multitenancy_enabled: true,
  private_tenant_enabled: true,
  default_tenant: '',
};

// eslint-disable-next-line
const mockTenantUtils = require('../../../utils/tenant-utils');
// eslint-disable-next-line
const mockAuthInfoUtils = require('../../../../../utils/auth-info-utils');

describe('Tenant list', () => {
  const setState = jest.fn();
  const mockCoreStart = {
    http: 1,
    chrome: {
      setBreadcrumbs() {
        return 1;
      },
    },
  };
  const config = {
    multitenancy: {
      enabled: true,
      tenants: {
        enable_private: true,
      },
    },
  };

  beforeEach(() => {
    jest.spyOn(React, 'useState').mockImplementation((initialValue) => [initialValue, setState]);
  });

  it('Render empty', () => {
    (getDashboardsInfo as jest.Mock).mockImplementation(() => {
      return mockDashboardsInfo;
    });

    const mockTenantListingData = [
      {
        tenant: 'tenant_1',
        description: '',
        reserved: true,
        tenantValue: 'tenant_1',
      },
    ];

    mockTenantUtils.transformTenantData = jest.fn().mockReturnValue(mockTenantListingData);

    const component = shallow(
      <ManageTab
        coreStart={mockCoreStart as any}
        navigation={{} as any}
        params={{} as any}
        config={config as any}
      />
    );

    expect(component.find(EuiInMemoryTable).prop('items').length).toBe(0);
  });

  it('renders when multitenancy is disabled in the opensearch_dashboards.yml', () => {
    (getDashboardsInfo as jest.Mock).mockImplementation(() => {
      return {
        multitenancy_enabled: false,
        private_tenant_enabled: true,
        default_tenant: '',
      };
    });
    const config1 = {
      multitenancy: {
        enabled: false,
        tenants: {
          enable_private: true,
        },
      },
    };
    const component = shallow(
      <ManageTab
        coreStart={mockCoreStart as any}
        navigation={{} as any}
        params={{} as any}
        config={config1 as any}
      />
    );
    expect(component.find(TenantInstructionView).length).toBe(1);
  });

  it('fetch data error', (done) => {
    jest.spyOn(React, 'useEffect').mockImplementationOnce((f) => f());
    // Hide the error message
    jest.spyOn(console, 'log').mockImplementationOnce(() => {});
    mockTenantUtils.fetchTenants.mockImplementationOnce(() => {
      throw Error();
    });

    shallow(
      <ManageTab
        coreStart={mockCoreStart as any}
        navigation={{} as any}
        params={{} as any}
        config={config as any}
      />
    );

    process.nextTick(() => {
      // Expect setting error to true
      expect(setState).toHaveBeenCalledWith(true);
      done();
    });
  });

  it('fetch data', (done) => {
    jest.spyOn(React, 'useEffect').mockImplementationOnce((f) => f());

    const mockTenantListingData = [
      {
        tenant: 'tenant_1',
        description: '',
        reserved: true,
        tenantValue: 'tenant_1',
      },
    ];

    mockTenantUtils.transformTenantData = jest.fn().mockReturnValue(mockTenantListingData);

    shallow(
      <ManageTab
        coreStart={mockCoreStart as any}
        navigation={{} as any}
        params={{} as any}
        config={config as any}
      />
    );

    process.nextTick(() => {
      expect(mockTenantUtils.fetchTenants).toHaveBeenCalledTimes(1);
      expect(mockTenantUtils.transformTenantData).toHaveBeenCalledTimes(1);
      expect(mockTenantUtils.fetchCurrentTenant).toHaveBeenCalledTimes(1);
      expect(mockAuthInfoUtils.getCurrentUser).toHaveBeenCalledTimes(1);
      expect(setState).toHaveBeenCalledWith(mockTenantListingData);
      done();
    });
  });

  it('delete tenant', (done) => {
    shallow(
      <ManageTab
        coreStart={mockCoreStart as any}
        navigation={{} as any}
        params={{} as any}
        config={config as any}
      />
    );
    const deleteFunc = useDeleteConfirmState.mock.calls[0][0];

    deleteFunc();

    process.nextTick(() => {
      expect(mockTenantUtils.requestDeleteTenant).toBeCalled();
      done();
    });
  });

  it('delete tenant error', (done) => {
    mockTenantUtils.requestDeleteTenant.mockImplementationOnce(() => {
      throw new Error();
    });
    // Hide the error message
    const loggingFunc = jest.fn();
    jest.spyOn(console, 'log').mockImplementationOnce(loggingFunc);
    shallow(
      <ManageTab
        coreStart={mockCoreStart as any}
        navigation={{} as any}
        params={{} as any}
        config={config as any}
      />
    );
    const deleteFunc = useDeleteConfirmState.mock.calls[0][0];

    deleteFunc();

    process.nextTick(() => {
      expect(loggingFunc).toBeCalled();
      done();
    });
  });

  it('submit tenant', () => {
    const component = shallow(
      <ManageTab
        coreStart={mockCoreStart as any}
        navigation={{} as any}
        params={{} as any}
        config={config as any}
      />
    );
    component.find('#createTenant').simulate('click');
    const submitFunc = component.find(TenantEditModal).prop('handleSave');
    submitFunc('tenant_1', '');

    expect(mockTenantUtils.updateTenant).toBeCalled();
  });

  it('submit tenant error', () => {
    const consoleLog = jest.spyOn(console, 'log');
    consoleLog.mockImplementation(() => {});
    const error = new Error();
    mockTenantUtils.updateTenant.mockImplementationOnce(() => {
      throw error;
    });
    const component = shallow(
      <ManageTab
        coreStart={mockCoreStart as any}
        navigation={{} as any}
        params={{} as any}
        config={config as any}
      />
    );
    component.find('#createTenant').simulate('click');
    const submitFunc = component.find(TenantEditModal).prop('handleSave');
    submitFunc('tenant_1', '');

    // Expect error log
    expect(consoleLog).toBeCalledWith(error);
  });

  describe('Action menu enable-disable check', () => {
    const sampleReservedTenant: Tenant = {
      tenant: 'tenant_1',
      description: '',
      reserved: true,
      tenantValue: 'tenant_1',
    };
    const sampleCustomTenant1: Tenant = {
      tenant: 'tenant_2',
      description: '',
      reserved: false,
      tenantValue: 'tenant_2',
    };
    const sampleCustomTenant2: Tenant = {
      tenant: 'tenant_3',
      description: '',
      reserved: false,
      tenantValue: 'tenant_3',
    };

    it('edit and delete should be disabled when selected tenant is reserved', () => {
      jest.spyOn(React, 'useState').mockImplementation(() => [[sampleReservedTenant], jest.fn()]);
      const component = shallow(
        <ManageTab
          coreStart={mockCoreStart as any}
          navigation={{} as any}
          params={{} as any}
          config={config as any}
        />
      );
      expect(component.find('#edit').prop('disabled')).toBe(true);
      expect(component.find('#delete').prop('disabled')).toBe(true);
    });

    it('All menues should be disabled when there is multiple tenant selected including reserved tenant', () => {
      jest
        .spyOn(React, 'useState')
        .mockImplementation(() => [[sampleReservedTenant, sampleCustomTenant1], jest.fn()]);
      const component = shallow(
        <ManageTab
          coreStart={mockCoreStart as any}
          navigation={{} as any}
          params={{} as any}
          config={config as any}
        />
      );
      expect(component.find('#switchTenant').prop('disabled')).toBe(true);
      expect(component.find('#edit').prop('disabled')).toBe(true);
      expect(component.find('#duplicate').prop('disabled')).toBe(true);
      expect(component.find('#createDashboard').prop('disabled')).toBe(true);
      expect(component.find('#createVisualizations').prop('disabled')).toBe(true);
      expect(component.find('#delete').prop('disabled')).toBe(true);
    });

    it('All menues should be disabled except delete when there is multiple custom tenant selected', () => {
      jest
        .spyOn(React, 'useState')
        .mockImplementation(() => [[sampleCustomTenant1, sampleCustomTenant2], jest.fn()]);
      const component = shallow(
        <ManageTab
          coreStart={mockCoreStart as any}
          navigation={{} as any}
          params={{} as any}
          config={config as any}
        />
      );
      expect(component.find('#switchTenant').prop('disabled')).toBe(true);
      expect(component.find('#edit').prop('disabled')).toBe(true);
      expect(component.find('#duplicate').prop('disabled')).toBe(true);
      expect(component.find('#createDashboard').prop('disabled')).toBe(true);
      expect(component.find('#createVisualizations').prop('disabled')).toBe(true);

      expect(component.find('#delete').prop('disabled')).toBe(false);
    });
  });

  describe('Action menu click', () => {
    let component;
    const sampleCustomTenant1: Tenant = {
      tenant: 'tenant_2',
      description: '',
      reserved: false,
      tenantValue: 'tenant_2',
    };

    beforeEach(() => {
      jest.spyOn(React, 'useState').mockImplementation(() => [[sampleCustomTenant1], jest.fn()]);
      component = shallow(
        <ManageTab
          coreStart={mockCoreStart as any}
          navigation={{} as any}
          params={{} as any}
          config={config as any}
        />
      );
    });

    it('switchTenant click', () => {
      component.find('#switchTenant').simulate('click');
      expect(mockTenantUtils.selectTenant).toBeCalled();
    });

    it('Edit click', () => {
      component.find('#edit').simulate('click');
      expect(component).toMatchSnapshot();
    });

    it('Duplicate click', () => {
      component.find('#duplicate').simulate('click');
      expect(component).toMatchSnapshot();
    });

    it('Create dashboard click', () => {
      component.find('#createDashboard').simulate('click');
      expect(mockTenantUtils.selectTenant).toHaveBeenCalled();
    });

    it('Create visualizations click', () => {
      component.find('#createVisualizations').simulate('click');
      expect(mockTenantUtils.selectTenant).toHaveBeenCalled();
    });
  });
});
