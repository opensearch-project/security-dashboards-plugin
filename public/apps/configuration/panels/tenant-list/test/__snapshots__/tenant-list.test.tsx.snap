// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tenant list AccessError component should load access error component: configure tab 1`] = `
<AccessErrorComponent
  dataSourceLabel="Local cluster"
  message="You do not have permissions to configure tenancy"
/>
`;

exports[`Tenant list AccessError component should load access error component: manage tab 1`] = `
<AccessErrorComponent
  dataSourceLabel="Local cluster"
  message="You do not have permissions to manage tenants"
/>
`;

exports[`Tenant list Action menu click Duplicate click 1`] = `
<Fragment>
  <EuiPageHeader />
  <EuiPageContent>
    <EuiPageContentHeader>
      <EuiPageContentHeaderSection>
        <EuiTitle
          size="s"
        >
          <h3>
            Dashboards tenants
            <span
              className="panel-header-count"
            >
               
              (
              1
              )
            </span>
          </h3>
        </EuiTitle>
        <EuiText
          color="subdued"
          size="s"
        >
          Manage tenants that already have been created. Global tenant is the default when tenancy is turned off.
        </EuiText>
      </EuiPageContentHeaderSection>
      <EuiPageContentHeaderSection>
        <EuiFlexGroup>
          <EuiFlexItem>
            <EuiButtonEmpty
              disabled={false}
              id="switchTenant"
              key="switchTenant"
              onClick={[Function]}
            >
              Switch to selected tenant
            </EuiButtonEmpty>
            <EuiButtonEmpty
              disabled={false}
              id="edit"
              key="edit"
              onClick={[Function]}
            >
              Edit
            </EuiButtonEmpty>
            <EuiButtonEmpty
              disabled={false}
              id="duplicate"
              key="duplicate"
              onClick={[Function]}
            >
              Duplicate
            </EuiButtonEmpty>
            <EuiButtonEmpty
              disabled={false}
              id="createDashboard"
              key="createDashboard"
              onClick={[Function]}
            >
              Create dashboard
            </EuiButtonEmpty>
            <EuiButtonEmpty
              disabled={false}
              id="createVisualizations"
              key="createVisualizations"
              onClick={[Function]}
            >
              Create visualizations
            </EuiButtonEmpty>
            <EuiButtonEmpty
              disabled={true}
              id="set as Global Default"
              key="Set as Global Default key"
              onClick={[Function]}
            >
              Set as Default Tenant
            </EuiButtonEmpty>
            <EuiButtonEmpty
              color="danger"
              disabled={false}
              id="delete"
              key="delete"
              onClick={[MockFunction]}
            >
              Delete
            </EuiButtonEmpty>
          </EuiFlexItem>
          <EuiFlexItem>
            <EuiButton
              fill={true}
              id="createTenant"
              onClick={[Function]}
            >
              Create tenant
            </EuiButton>
          </EuiFlexItem>
        </EuiFlexGroup>
      </EuiPageContentHeaderSection>
    </EuiPageContentHeader>
    <EuiPageBody>
      <EuiInMemoryTable
        columns={
          Array [
            Object {
              "field": "tenant",
              "name": "Name",
              "render": [Function],
              "sortable": true,
            },
            Object {
              "field": "tenant",
              "name": "Status",
              "render": [Function],
            },
            Object {
              "field": "description",
              "name": "Description",
              "truncateText": true,
            },
            Object {
              "field": "tenantValue",
              "name": "Dashboard",
              "render": [Function],
            },
            Object {
              "field": "tenantValue",
              "name": "Visualizations",
              "render": [Function],
            },
            Object {
              "field": "reserved",
              "name": "Customization",
              "render": [Function],
            },
          ]
        }
        error=""
        itemId="tenant"
        items={
          Array [
            Object {
              "description": "",
              "reserved": false,
              "tenant": "tenant_2",
              "tenantValue": "tenant_2",
            },
          ]
        }
        loading={false}
        message={false}
        pagination={true}
        responsive={true}
        search={
          Object {
            "box": Object {
              "placeholder": "Find tenant",
            },
            "onChange": [Function],
          }
        }
        selection={
          Object {
            "onSelectionChange": [MockFunction],
          }
        }
        sorting={true}
        tableLayout="auto"
      />
    </EuiPageBody>
  </EuiPageContent>
  <TenantEditModal
    action="duplicate"
    handleClose={[Function]}
    handleSave={[Function]}
    tenantDescription=""
    tenantName="tenant_2_copy"
  />
  <EuiGlobalToastList
    dismissToast={[MockFunction]}
    side="right"
    toastLifeTimeMs={10000}
    toasts={Array []}
  />
</Fragment>
`;

exports[`Tenant list Action menu click Edit click 1`] = `
<Fragment>
  <EuiPageHeader />
  <EuiPageContent>
    <EuiPageContentHeader>
      <EuiPageContentHeaderSection>
        <EuiTitle
          size="s"
        >
          <h3>
            Dashboards tenants
            <span
              className="panel-header-count"
            >
               
              (
              1
              )
            </span>
          </h3>
        </EuiTitle>
        <EuiText
          color="subdued"
          size="s"
        >
          Manage tenants that already have been created. Global tenant is the default when tenancy is turned off.
        </EuiText>
      </EuiPageContentHeaderSection>
      <EuiPageContentHeaderSection>
        <EuiFlexGroup>
          <EuiFlexItem>
            <EuiButtonEmpty
              disabled={false}
              id="switchTenant"
              key="switchTenant"
              onClick={[Function]}
            >
              Switch to selected tenant
            </EuiButtonEmpty>
            <EuiButtonEmpty
              disabled={false}
              id="edit"
              key="edit"
              onClick={[Function]}
            >
              Edit
            </EuiButtonEmpty>
            <EuiButtonEmpty
              disabled={false}
              id="duplicate"
              key="duplicate"
              onClick={[Function]}
            >
              Duplicate
            </EuiButtonEmpty>
            <EuiButtonEmpty
              disabled={false}
              id="createDashboard"
              key="createDashboard"
              onClick={[Function]}
            >
              Create dashboard
            </EuiButtonEmpty>
            <EuiButtonEmpty
              disabled={false}
              id="createVisualizations"
              key="createVisualizations"
              onClick={[Function]}
            >
              Create visualizations
            </EuiButtonEmpty>
            <EuiButtonEmpty
              disabled={true}
              id="set as Global Default"
              key="Set as Global Default key"
              onClick={[Function]}
            >
              Set as Default Tenant
            </EuiButtonEmpty>
            <EuiButtonEmpty
              color="danger"
              disabled={false}
              id="delete"
              key="delete"
              onClick={[MockFunction]}
            >
              Delete
            </EuiButtonEmpty>
          </EuiFlexItem>
          <EuiFlexItem>
            <EuiButton
              fill={true}
              id="createTenant"
              onClick={[Function]}
            >
              Create tenant
            </EuiButton>
          </EuiFlexItem>
        </EuiFlexGroup>
      </EuiPageContentHeaderSection>
    </EuiPageContentHeader>
    <EuiPageBody>
      <EuiInMemoryTable
        columns={
          Array [
            Object {
              "field": "tenant",
              "name": "Name",
              "render": [Function],
              "sortable": true,
            },
            Object {
              "field": "tenant",
              "name": "Status",
              "render": [Function],
            },
            Object {
              "field": "description",
              "name": "Description",
              "truncateText": true,
            },
            Object {
              "field": "tenantValue",
              "name": "Dashboard",
              "render": [Function],
            },
            Object {
              "field": "tenantValue",
              "name": "Visualizations",
              "render": [Function],
            },
            Object {
              "field": "reserved",
              "name": "Customization",
              "render": [Function],
            },
          ]
        }
        error=""
        itemId="tenant"
        items={
          Array [
            Object {
              "description": "",
              "reserved": false,
              "tenant": "tenant_2",
              "tenantValue": "tenant_2",
            },
          ]
        }
        loading={false}
        message={false}
        pagination={true}
        responsive={true}
        search={
          Object {
            "box": Object {
              "placeholder": "Find tenant",
            },
            "onChange": [Function],
          }
        }
        selection={
          Object {
            "onSelectionChange": [MockFunction],
          }
        }
        sorting={true}
        tableLayout="auto"
      />
    </EuiPageBody>
  </EuiPageContent>
  <TenantEditModal
    action="edit"
    handleClose={[Function]}
    handleSave={[Function]}
    tenantDescription=""
    tenantName="tenant_2"
  />
  <EuiGlobalToastList
    dismissToast={[MockFunction]}
    side="right"
    toastLifeTimeMs={10000}
    toasts={Array []}
  />
</Fragment>
`;
